### PR TITLE
Add furniture problem

### PR DESCRIPTION
**Pull request by Paula-Kli**
When trying to build the pdf I get the error, that I am not able to locate template.tex. Thus I don't know whether actually it looks well.
I will try to make it work but if you notice some problems please let me know

### CODE REVIEW

File: cp22/hpi-fun2/furniture/problem.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "description": "Given parts of furniture and their dependencies to one another return the time needed to build them.",
```
Comment (schmidtjonas) [inline cp22/hpi-fun2/furniture/problem.json]: We usually write something about the solution, like 
```suggestion
  "description": "longest path in DAG with vertex weights and story",
```

File: cp22/hpi-fun2/furniture/problem.tex [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+All tasks that are before one task need to be finished before a new one can be started.
+
+
+\paragraph*{Input}
+
+The first line contains $n$ $(1\leq n \leq 10^4)$ the number of steps that need to be done to assemble that piece of furniture.
+
+Then $n$ numbers $t$ $(1\leq t \leq 100)$ follow which represent the time it takes to finish each step.
+
+The next line contains $m$ the number of instructions of which step need to be completed before the next one starts. Followed
```
Comment (schmidtjonas) [inline cp22/hpi-fun2/furniture/problem.tex]: please add a bound for m (something around 10^5)

File: cp22/hpi-fun2/furniture/executables/generator.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+
+def generate_random_testcase(number: int):
+    split = math.ceil(number / 3 * 2) + 1
+    edges = [(i - 1, i) for i in range(2, split)]
+    j = 1
+    for i in range(split, number + 1):
+        edges += [(j, i)]
+        edges += [(i, j + 2)]
+        j += 2
```
Comment (schmidtjonas) [inline cp22/hpi-fun2/furniture/executables/generator.py]: I think it's possible that this does not create a dag. Please check again.
To create random dags, why not generate a random toposort (using shuffle) and add random edges in the correct direction.

**Comment by schmidtjonas** [comment]
Regarding the bounds: n can easily be 10^5 and m should also get a bound (like 2*10^5). The bound for t is very small, but that can be a nice bait :)

**Comment by schmidtjonas** [comment]
The pdf looks good, not sure why you get an error.

**Comment by schmidtjonas** [comment]
Note that a bfs-dp should be wrong. You have to do the DP in order of the topological sorting like in your solution. This is wrong in the notes

File: cp22/hpi-fun2/furniture/problem.tex [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+by $m$ lines containing instructions $a$ and $b$ saying step $a$ needs to be finished before $b$ can start.
```
Comment (schmidtjonas) [inline cp22/hpi-fun2/furniture/problem.tex]: please add bounds for a and b to see if they are 1-indexed

**Comment by schmidtjonas** [comment]
My solution works on the samples, but the assertion for the graph to be a DAG fails

**Comment by schmidtjonas** [comment]
> Note that a bfs-dp should be wrong. You have to do the DP in order of the topological sorting like in your solution. This is wrong in the notes

Feel free to ping me once that is fixed, so that I can animate and fix your ipe slide :)

**Comment by schmidtjonas** [comment]
Also: please remove the `notes.md` file in the end, if you have a `notes.ipe`

**Comment by Paula-Kli** [comment]
> Regarding the bounds: n can easily be 10^5 and m should also get a bound (like 2*10^5). The bound for t is very small, but that can be a nice bait :)

Taking those bounds my solution in python exceeds the one second time limit on my machine ... Shall I increase the bounds anyways?

File: cp22/hpi-fun2/furniture/executables/generator.py (hunk 3 above)
Comment (Paula-Kli) [reply cp22/hpi-fun2/furniture/executables/generator.py]: I think that I created a better solution for this even though @schmidtjonas your solution still fails on some testcases

**Comment by schmidtjonas** [comment]
Looks all good now. I approve

**Pull request merged**